lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/domain/CriResponse.java

```diff
@@ -2,14 +2,20 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import uk.gov.di.ipv.core.library.domain.JourneyResponse;
 
 @Getter
-public class CriResponse {
+@EqualsAndHashCode(callSuper = true)
+public class CriResponse extends JourneyResponse {
```

Comment:
I agree that a generic approach would be better. I will pick up where remove the inheritance of 'BaseJourneyLambda' from the subtask 'Update the Lambda to support both API Gateway and JSON request'. (PYIC-2750) In some other Lambdas, we are returning object then 'JourneyResponse,' and they are also extended from 'JourneyResponse'. (For example: build-client-auth-response) If we are not going to follow that approach, I will consider making it generic while extending from 'RequestHandler' in all Lambdas, keeping this in mind. However, It would be more appropriate to fix this in other Lambdas as well and follow same approach.

Reply:
Yes this'll need to be fixed in all the lambdas.

Reply:
implemented the interface as we have decided.